### PR TITLE
[TRIGGER] Stripe #6999

### PR DESCRIPTION
**Pull request by lcaresia**
### <samp>🤖 Generated by Copilot at a4bab0a</samp>

Refactored and added new stripe webhook sources. The pull request introduces a common `webhook-base.mjs` file that contains the shared logic and properties for the stripe webhook sources. It also adds 10 new sources that extend the base object and emit events for different stripe events, such as new customers, invoices, payments, disputes, subscriptions, etc. The pull request also updates the stripe package version in `package.json`.

### <samp>🤖 Generated by Copilot at a4bab0a</samp>

> _Stripe webhooks grow_
> _`webhook-base` is the root_
> _Autumn of refactor_

### <samp>🤖 Generated by Copilot at a4bab0a</samp>

*  Add a common webhook base object for stripe sources (`webhook-base.mjs`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-c1e7be45c237dc62b5c18a50fc5f04660ce9c684d10b4bae46aaf6bd99dbf5d3R1-R93))
*  Update the custom webhook events source to use the common webhook base object (`custom-webhook-events.mjs`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-d2170cfd7b3aea606f13850cbd1feddf1beb6c12c73c002aa40ef556dcf1d372L1-R12), [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-d2170cfd7b3aea606f13850cbd1feddf1beb6c12c73c002aa40ef556dcf1d372L21-R27))
*  Add new sources for various stripe events that extend the common webhook base object and override the relevant props and methods (`new-canceled-subscription.mjs`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-7b2e980d08e7879475725fd35dcc045c895e2e18246f3f35c32c2e9c22bcfb49R1-R18); `new-customer.mjs`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-afcc97ea886b3dd872289b7433924962c24c0cfde65fa232e56fb63b936d22e4R1-R18); `new-dispute.mjs`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-61a3c84f698a4b778ba7625a85d9e935c8a1952278ca51ccea326bd68e5e4b6eR1-R19); `new-failed-invoice-payment.mjs`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-05fdde8bad65ba04574134d64614ccd37bf5f6a6ab0b33ee585b44c23134c828R1-R18); `new-failed-payment.mjs`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-32f87340ab910936f073e170657d8dc841ea1290f25617552e95e9a1e9edf589R1-R19); `new-invoice.mjs`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-f5c2bb2f4c3cfa44d9117fbc5da8679094b2fa48d6b166128bbcacd8614c7629R1-R18); `new-payment.mjs`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-a6bb01709eec66477d6745b21ce434d646dd448566fe31552a0136f805fc8e9bR1-R18); `new-subscription-updated.mjs`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-c6cbd3f9942960c8cd6bb4908cc3ccf54b86b0ee2b586ecd0158c5edb99ce696R1-R18); `new-subscription.mjs`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-9239481919098ec330ecf2640362d6f23c5107512f5ffa7f94bf070378e96428R1-R19))
*  Bump up the version of the stripe package to reflect the new features and changes (`package.json`, [link](https://github.com/PipedreamHQ/pipedream/pull/7029/files?diff=unified&w=0#diff-731f6694b62d00bab34491f9eed82a3638d50935d54fc32fb3abe6d91db3fcd2L3-R3))

### CODE REVIEW

File: components/stripe/sources/new-failed-invoice-payment/new-failed-invoice-payment.mjs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import common from "../common/webhook-base.mjs";
+
+export default {
+  ...common,
+  key: "stripe-new-failed-invoice-payment",
+  name: "New Failed Invoice Payment",
```
Comment (GTFalcao) [inline components/stripe/sources/new-failed-invoice-payment/new-failed-invoice-payment.mjs:6]: Invoice Payment Failed

File: components/stripe/sources/new-failed-payment/new-failed-payment.mjs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import common from "../common/webhook-base.mjs";
+
+export default {
+  ...common,
+  key: "stripe-new-failed-payment",
+  name: "New Failed Payment",
```
Comment (GTFalcao) [inline components/stripe/sources/new-failed-payment/new-failed-payment.mjs:6]: Payment Failed

File: components/stripe/sources/new-subscription-updated/new-subscription-updated.mjs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import common from "../common/webhook-base.mjs";
+
+export default {
+  ...common,
+  key: "stripe-new-subscription-updated",
+  name: "New Subscription Updated",
```
Comment (GTFalcao) [inline components/stripe/sources/new-subscription-updated/new-subscription-updated.mjs:6]: Subscription Updated

File: components/stripe/sources/new-canceled-subscription/new-canceled-subscription.mjs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import common from "../common/webhook-base.mjs";
+
+export default {
+  ...common,
+  key: "stripe-new-canceled-subscription",
+  name: "New Canceled Subscription",
```
Comment (GTFalcao) [inline components/stripe/sources/new-canceled-subscription/new-canceled-subscription.mjs:6]: Subscription Canceled

**Review by GTFalcao — COMMENTED** [review]
Hi @lcaresia , thanks for the contribution!

I left some suggestions regarding the source names - perhaps @dannyroosevelt @malexanderlim could provide their opinions on this as well. I know we usually begin source names with "New" and we even have a custom lint suggestion for that, but for sources that don't actually relate to new items and rather trigger on events ocurring on existing ones, I don't think we have to add that.

If we do change them, the keys and descriptions should be adjusted accordingly.

That being said, moving this to QA since these possible changes have nothing to do with the actual code running.

**Comment by dannyroosevelt** [comment]
> If we do change them, the keys and descriptions should be adjusted accordingly.

Great suggestion, @GTFalcao. I'll comment on each specifically.

File: components/stripe/sources/new-failed-invoice-payment/new-failed-invoice-payment.mjs (hunk 1 above)
Comment (dannyroosevelt) [reply components/stripe/sources/new-failed-invoice-payment/new-failed-invoice-payment.mjs:6]: I think this one is ok actually -- "New Failed Invoice Payment" makes sense.

File: components/stripe/sources/new-failed-payment/new-failed-payment.mjs (hunk 2 above)
Comment (dannyroosevelt) [reply components/stripe/sources/new-failed-payment/new-failed-payment.mjs:6]: I think this one is ok as well -- "New Failed Payment" makes sense.

**Review by GTFalcao — APPROVED** [review]

**Review by GTFalcao — APPROVED** [review]

**Pull request merged**